packages/runtime/src/ctx.ts

```diff
@@ -747,6 +753,9 @@ export class Ctx implements Context {
         }
       }
 
+      // apply rekey after pass all logic
+      this.rekeyTo(txParam);
```

Comment:
cool, should work. Could you please also check few direct API's which we have (which doesn't use processTransactions function)? If not, we're good

Reply:
We have some APIs like runtime.{deployApp, updateApp..., } which parameters didn't provide the `rekey` field. So I think we will do it in other PR and ensure Runtime and Algob.Deployer are compatible.

Reply:
> cool, should work. Could you please also check few direct API's which we have (which doesn't use processTransactions function)? If not, we're good

We have some APIs like runtime.{deployApp, updateApp..., } which parameters didn't provide the rekey field. So I think we will do it in other PR and ensure Runtime and Algob.Deployer are compatible.